File: GameStats.React/ClientApp/src/DTOs/Coach/CoachDto.js
export default class CoachDto {
    constructor(options = {}) {
        this.id = options.id || 0;
        this.firstName = options.firstName || '';
        this.lastName = options.lastName || '';
        this.isHeadCoach = options.isHeadCoach;
        this.teamId = options.teamId || 0;

        this.name = `${this.firstName} ${this.lastName}`;
    }

    stringify() {
        return JSON.stringify(this);
    }
}

File: GameStats.Common/Contracts/Coach/CoachDto.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace GameStats.Common.Contracts.Coach
{
    public class CoachDto
    {
        [JsonPropertyName("id")]
        public int Id { get; set; }

        [JsonPropertyName("firstName")]
        public string FirstName { get; set; }

        [JsonPropertyName("lastName")]
        public string LastName { get; set; }

        [JsonPropertyName("isHeadCoach")]
        public bool IsHeadCoach { get; set; }

        [JsonPropertyName("TeamId")]
        public int TeamId { get; set; }
    }

    public class CoachListDto
    {
        public List<CoachDto> Coaches { get; set; } = new();
    }
}
